aws-push-notifications-pinpoint-common/src/main/java/com/amplifyframework/pushnotifications/pinpoint/PinpointNotificationPayload.kt

```diff
@@ -17,42 +17,29 @@ package com.amplifyframework.pushnotifications.pinpoint
 
 import androidx.annotation.RestrictTo
 import com.amplifyframework.notifications.pushnotifications.NotificationPayload
-import com.amplifyframework.pushnotifications.pinpoint.utils.PushNotificationsConstants
-import com.google.firebase.messaging.RemoteMessage
-import kotlin.collections.HashMap
 
-@RestrictTo(RestrictTo.Scope.LIBRARY_GROUP)
+@RestrictTo(RestrictTo.Scope.LIBRARY)
 class PinpointNotificationPayload internal constructor(
-    messageId: String? = null,
-    senderId: String? = null,
-    sendTime: Long? = null,
-    title: String? = null,
-    body: String? = null,
-    imageUrl: String? = null,
-    channelId: String? = null,
-    action: Map<String, String> = mapOf(),
-    silentPush: Boolean = false,
-    rawData: Map<String, String> = mapOf()
-) : NotificationPayload(
-    messageId, senderId, sendTime,
-    title, body, imageUrl,
-    channelId, action, silentPush, rawData
-) {
+    var title: String? = null,
+    var body: String? = null,
+    var imageUrl: String? = null,
+    var action: Map<String, String> = mapOf(),
+    var silentPush: Boolean = false
+) : NotificationPayload() {
     companion object {
-        fun createFromRemoteMessage(remoteMessage: RemoteMessage): PinpointNotificationPayload? {
-            if (!remoteMessage.data.keys.any { it.contains(PushNotificationsConstants.PINPOINT_PREFIX) }) return null
+        fun isPinpointNotificationPayload(payload: NotificationPayload) = payload.rawData.keys.any {
+            it.contains(PushNotificationsConstants.PINPOINT_PREFIX)
+        }
 
-            val data = remoteMessage.data
-            val messageId = remoteMessage.messageId
-            val senderId = remoteMessage.senderId
-            val sendTime = remoteMessage.sentTime
+        fun fromNotificationPayload(payload: NotificationPayload): PinpointNotificationPayload {
+            val data = payload.rawData
             val title = data[PushNotificationsConstants.TITLE]
                 ?: data[PushNotificationsConstants.PINPOINT_NOTIFICATION_TITLE]
             val body = data[PushNotificationsConstants.MESSAGE]
                 ?: data[PushNotificationsConstants.PINPOINT_NOTIFICATION_BODY]
             val imageUrl = data[PushNotificationsConstants.IMAGEURL]
                 ?: data[PushNotificationsConstants.PINPOINT_NOTIFICATION_IMAGEURL]
-            val channelId = PushNotificationsConstants.DEFAULT_NOTIFICATION_CHANNEL_ID
+            val channelId = payload.channelId ?: PushNotificationsConstants.DEFAULT_NOTIFICATION_CHANNEL_ID
```

Comment:
Doesn't this need an override on NotificationPayload value?

Reply:
Added override. But we still need default channel if payload channel is null.